core/src/test/java/org/apache/calcite/test/MockSqlOperatorTable.java

```diff
@@ -125,6 +125,18 @@ public RelDataType inferReturnType(SqlOperatorBinding opBinding) {
     }
   }
 
+  /** "MYAGGFUNC" user-defined aggregate function. This agg function accept one or more arguments
+   * in order to reproduce the throws of CALCITE-3929. */
+  public static class MyAggFunc extends SqlAggFunction {
```

Comment:
extends `SqlUserDefinedAggFunction` might be better for your case

Reply:
There is a `MockSqlOperatorTable#MyAvgAggFunction`, we should use it to reproduce this case.

Reply:
@yanlin-Lynn
Thanks, `SqlUserDefinedAggFunction`  has an AggregateFunction argument, it's not convenient for reproduce this case.So it's better to extends SqlAggFunction and define it as a user-defined function.